pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -236,6 +240,40 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, img string, saNam
 				Value: grpcPodConfig.MemoryTarget.String() + "B", // k8s resources use Mi, GOMEMLIMIT wants MiB
 			})
 		}
+
+		// Reconfigure pod to extract content
+		if grpcPodConfig.ExtractContent != nil {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: "catalog-content",
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			})
+			const catalogPath = "/catalog"
+			volumeMount := corev1.VolumeMount{
+				Name:      "catalog-content",
+				MountPath: catalogPath,
+			}
+			pod.Spec.InitContainers = append(pod.Spec.InitContainers, corev1.Container{
+				Name:    "extract-content",
+				Image:   img,
+				Command: []string{"/bin/sh", "-c"},
+				Args: []string{fmt.Sprintf("cp -r %s %s/catalog && cp -r %s %s/cache",
+					grpcPodConfig.ExtractContent.CatalogDir, catalogPath,
+					grpcPodConfig.ExtractContent.CacheDir, catalogPath,
+				)},
```

Comment:
This requires the FBC catalog image to have `/bin/sh`, right?

Is there different way to do this, where we:
- initContainer: copy a `cp` binary or equivalent into an emptyDir called `copy-binary`
- initContainer(catalog image): mount `copy-binary`, run `cp` to copy catalog and cache into another emptydir called `catalog-content`
- container(opm): mount `catalog-content`, run `opm serve` pointing to mounted content dirs.

That way, we make no assumptions about the contents of the catalog image, other than the catalog and cache dirs.

Reply:
> That way, we make no assumptions about the contents of the catalog image, other than the catalog and cache dirs.

That leads me to more questions:
- Do we need to go back to the api and make `catalogDir` required?
- If `cacheDir` is unset, (FBC images don't have to have caches), do we handle that appropriately here?

Reply:
Why do you think it's more onerous to expect `sh` and `cp` than it is to expect FBC? We certainly could use some other image (which ... ?) to inject them, but it's not clear to me what niche that targets.

Reply:
> https://github.com/operator-framework/operator-lifecycle-manager/pull/3029#discussion_r1320139215

Both are required (optional is opt-in, right?)

Why should we bother to handle the case where the cache is missing? I'd rather enforce that everyone has one if they want to use this feature.

Reply:
Yikes, didn't realize my `omitempty` autocomplete had made them implciitly optional. Marking as required: https://github.com/operator-framework/api/pull/295

Reply:
> Why do you think it's more onerous to expect sh and cp

Those boil down to architecture-specific binaries/libraries being in the image, right? Which means we would have to keep building multi-arch catalog images. It would be _much_ nicer, IMO if we could have noarch `scratch`-based images that are literally just the catalogs and cache.

> Why should we bother to handle the case where the cache is missing? I'd rather enforce that everyone has one if they want to use this feature.

I'm not that fussed about it. If it starts off required, we can always come back later and make it optional if we need to.

Reply:
> We certainly could use some other image (which ... ?) 

Maybe the `opm` image or the OLM image? But that's a good question. For bundle unpacking, OLM has its own `cpb` binary that it uses. Not sure how easy it would be to copy over the standard `cp` executable and its dependencies.

Reply:
Let's start with the `opm` image and add additional configuration to the catalog operator as we see fit.